### ISSUE TITLE
Add parameter to monitors to allow execution every_n_epochs

### ISSUE THREAD

**Issue by andreas-eberle** [tensorflow/tensorflow#8361]
Many of the Monitors used for monitoring the training with an Estimator allow to specify the number of training steps after which they will be executed. Additionally it would be nice to be able to specify the number of epochs after which the monitor should be executed. This would make it very easy to validate the training after all training data has been learned for example one (or two...) more times. 

From looking at https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/learn/python/learn/monitors.py, I saw that all the monitors allowing to specify the `every_n_steps` parameter are extending the EveryN class. In order to also allow a parameter like `every_n_epochs`, this class could be extended to count the number of epochs by listening to the `def epoch_begin(self, epoch)` method.

**Comment by concretevitamin** [comment]
@ispirmustafa - do we have a plan to add this?  If not, we can mark this as contributions welcome.

**Comment by ispirmustafa** [comment]
No. This is not in our plan. Epoch doesn't work in distributed settings.
BTW, monitores are deprecated. Please use tf.train.SessionRunHook

**Comment by tomwanzek** [comment]
@ispirmustafa I know, this is drifting slightly from the specific issue at hand here, however... If, as you pointed out, the monitors API is deprecated, is there any effort on the way to update the core tutorial to reflect best practices in using `tf.train.SessionRunHook`? I saw related discussions in #7669.

Also, I did not see readily see an indication in the Release Notes. Thx, in advance for an indication.
